tests/system/Validation/CreditCardRulesTest.php

```diff
@@ -1220,32 +1205,26 @@ public function creditCardProvider()
     /**
      * Used to generate fake credit card numbers that will still pass the Luhn
      * check used to validate the card so we can be sure the cards are recognized correctly.
-     *
-     * @return string
      */
-    protected function generateCardNum(int $prefix, int $length)
+    private function generateCardNum(int $prefix, int $length): string
     {
         $pos        = mb_strlen($prefix);
         $finalDigit = 0;
         $sum        = 0;
 
-        // Fill in the first values of the string based on $prefix
         $string = str_split($prefix);
-
-        // Pad out the array to the appropriate length
         $string = array_pad($string, $length, 0);
 
-        // Fill all of the remaining values with random numbers, except the last one.
         while ($pos < $length - 1) {
             $string[$pos++] = random_int(0, 9);
         }
 
-        // Calculate the Luhn checksum of the current values.
```

Comment:
This comment helps code readers. I hope not to remove.

Reply:
Or is it better that extracting method to calculate the Luhn checksum?
Then the code would be like:
```php
$string[--$length] = $this->calculateLuhnChecksum(...);
```

Reply:
Extracted